src/libxtp/bulkesp.cc

```diff
@@ -0,0 +1,768 @@
+//                    DMAT_tot = DMAT_tot - DMAT[0] + DMAT[1];
+//                }
+//                // Ground state + hole_contribution + electron contribution
+//            } else throw std::runtime_error("State entry not recognized");
+//
+//
+//            return (DMAT_tot);
+//        }
+
+        ub::matrix<double> Bulkesp::BuildOverlapMat(Orbitals &_molOrb, Orbitals &_globalOrb) {
+            ub::matrix<double> ov;
+            return (ov);
+        }
+
+        /**
+         * 
+         * @param _global_atomlist
+         * @param _local_atomlist
+         * @param _local_atomIndeces Indeces of atoms in current molecule
+         * @param _global_dmat
+         * @param _global_basis
+         * @param bs
+         * @param gridsize
+         * @param _grid
+         * @return 
+         */
+        ub::vector<double> Bulkesp::ComputeESP(std::vector< ctp::QMAtom* > & _global_atomlist,
+                std::vector< ctp::QMAtom* > & _local_atomlist, std::vector<unsigned> _local_atomIndeces,
+                ub::matrix<double> &_global_dmat, AOBasis &_global_basis, BasisSet &bs, string gridsize, Grid &_grid, double &netcharge) {
+
+
+            ub::vector<double> _ESPatGrid = ub::zero_vector<double>(_grid.getsize());
+
+            NumericalIntegrationPeriodic numway;
+
+            //numway.GridSetup(gridsize,&bs,_global_atomlist);
+            numway.setBox(boxLen*tools::conv::ang2bohr);
+            numway.SetRelevantAtomIds(_local_atomIndeces);
+            CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Entering Bulkesp::ComputeESP()"<< flush;
+            numway.GridSetup(gridsize, &bs, _local_atomlist, &_global_basis);
+            CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Calculate Potentials at Numerical Grid with gridsize " << gridsize << flush;
+            //As long as basis functions are well supported and molecules are smaller than 0.5*boxLen along any axis, then
+            //density integration should be accurate enough without making it explicitly periodic
+            //double N = numway.IntegrateDensity_Molecule(_global_dmat, &_global_basis, _local_atomIndeces);
+            double N = numway.IntegrateDensity(_global_dmat);
+            CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Calculated Potentials at Numerical Grid, Number of electrons is " << N << flush;
+
+            _do_round = false; //do not round net charge, we expect total charge to be non-int, as molecules can transfer some between themselves
+            netcharge = getNetcharge(_local_atomlist, N, false); //don't round
+
+            CTP_LOG(ctp::logDEBUG, *_log) << ctp::TimeStamp() << " Calculating ESP at CHELPG grid points" << flush;
+            //boost::progress_display show_progress( _grid.getsize() );
+
+
+            //#define MADELUNG_TEST
+#ifdef MADELUNG_TEST
+
+
+            double exactMadelung = 1.74756459463318;
+
+#define GROLIKE
```

Comment:
What does this part of the code do?

Reply:
It tests if the an older version of the Ewald implementation could get the right value of the Madelung constant. It's all inside an ifdef and should be removed. Instead I need to write a proper unit test for it.